### PR TITLE
wip: try to remove the `@overlaypass` macro

### PR DESCRIPTION
**Pull request by aviatesk**

### CODE REVIEW

File: src/CassetteOverlay.jl [hunk 1]
```diff
@@ -34,7 +28,8 @@ end
 
 function overlay_generator(passtype, fargtypes)
     tt = to_tuple_type(fargtypes)
-    match = _which(tt; method_table=method_table(passtype), raise=false)
+    mt = Base.invoke_in_world(typemax(UInt), method_table, passtype)
```
Comment (aviatesk) [inline src/CassetteOverlay.jl:31]: Unfortunately this doesn't seem to work -- it still results in a world age error.
I confirmed `invokelatest` does the same.

@Keno did you have something different in your mind?

Comment (Keno) [reply src/CassetteOverlay.jl:31]: no, I expected this to work

Comment (Keno) [reply src/CassetteOverlay.jl:31]: I guess we specifically disallow `invokelatest` in generated functions ...

Comment (aviatesk) [reply src/CassetteOverlay.jl:31]: So maybe `typemax(UInt)-1`? 😎

Comment (Keno) [reply src/CassetteOverlay.jl:31]: No, we disallow switching worlds inside of generated functions entirely, which is probably sensible if a little heavy handed.

Comment (aviatesk) [reply src/CassetteOverlay.jl:31]: Well, I guess `invoke_in_world` seems to be disallowed also.

**Pull request closed without merging**